test/unittest/tests/block/test_extent_list_insert_wo_block.cpp

```diff
@@ -0,0 +1,197 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * block_ext.c: [extent_list2] Test extent list functions part 2. (More to come.)
+ *
+ * Test insert functions without block: __block_ext_insert, and __block_off_insert.
+ */
+
+/* Choose one */
+#define DEBUG /* Print debugging output */
+//#undef DEBUG
+
+#include <algorithm>
+#include <memory>
+
+#include <catch2/catch.hpp>
+
+#include "test_util.h"
+#include "../utils.h"
+#include "../utils_extlist.h"
+#include "../wrappers/mock_session.h"
+#include "wt_internal.h"
+
+using namespace utils;
+
+/*!
+ * To sort off_size by _off and _size
+ */
+struct {
+    bool
+    operator()(const off_size &left, const off_size &right)
+    {
+        return (left < right);
+    }
+} off_size_Less_off_and_size;
+
+TEST_CASE("Extent Lists: block_ext_insert", "[extent_list2]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> mock_session = MockSession::buildTestMockSession();
+    WT_SESSION_IMPL *session = mock_session->getWtSessionImpl();
+
+    std::vector<WT_EXT **> stack(WT_SKIP_MAXDEPTH, nullptr);
+
+    SECTION("insert into an empty list has one element")
+    {
+        BREAK;
+        /* Setup */
+        /* Empty extent list */
+        WT_EXTLIST extlist;
+        memset(&extlist, 0, sizeof(extlist));
+        verify_empty_extent_list(&extlist.off[0], &stack[0]);
+
```

Comment:
Each of these sections starts with a similar block of code:
```
        WT_EXTLIST extlist;
        memset(&extlist, 0, sizeof(extlist));
        verify_empty_extent_list(&extlist.off[0], &stack[0]);
```
with 
`        extlist_free(session, extlist);`
at the end of the sections.
It would be nice to be able reuse that code. A couple of options spring to mind: either or both of:

1. create a wrapper class that contains the `WT_EXTLIST`, initialises it and cleans it up, while providing an accessor method to make the `WT_EXTLIST` available, and/or
2. putting the creation and cleanup outside the `SECTION`s as part of the setup/teardown code so that only one copy is needed.

Reply:
1. With replacement of memset() with initialization and removal of the verify_empty_extent_list() which is unnecessary since it is already done in test_extent_list.cpp this reduces three lines to one line:

     WT_EXTLIST extlist = {};

I no longer think any simplification is needed.

Reply:
Hi @tod-johnson-mongodb , I agree that your simplification 
`WT_EXTLIST extlist = {};`
is a good change, and nothing further is needed